### PR TITLE
[RFR] Allow image removal

### LINKED ISSUE #349: ImageInput keeps adding imagest to state
Hi there.

In ImageInput's sources there are lines in onDrop function
`const updatedFiles = [
            ...this.state.files,
            ...files.map(this.transformFile),
        ];`

Which keeps adding images to state and there is no way to remove unwanted ones.
You might see in this a problem too.

Thanks.

### PR DESCRIPTION
**Pull request by jpetitcolas**
- [x] Add remove button to pictures
- [x] Add tests
- [x] Improve UI with decreased opacity unless hovering the picture

Add a button to remove undesired images from `<ImageInput />`.

anim

Fixes #349.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,6 +42,7 @@
     "eslint-plugin-react": "~6.7.1",
     "express": "^4.14.0",
     "extract-text-webpack-plugin": "~1.0.1",
+    "file-api": "^0.10.4",
```
Comment (djhi) [inline package.json]: `^` spotted

File: src/mui/input/ImageInputPreview.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import React, { PropTypes } from 'react';
+import FlatButton from 'material-ui/FlatButton';
+import { red600 } from 'material-ui/styles/colors';
+import RemoveCircle from 'material-ui/svg-icons/content/remove-circle';
+
+
```
Comment (djhi) [inline src/mui/input/ImageInputPreview.js]: extra line

**Comment by jpetitcolas** [comment]
Code review applied, and style of remove button updated: 

anim

I use now the same color as the top delete button.

File: example/addUploadFeature.js [hunk 3]
```diff
@@ -9,15 +9,20 @@ const convertFileToBase64 = file => new Promise((resolve, reject) => {
 const addUploadCapabilities = requestHandler => (type, resource, params) => {
     if (type === 'UPDATE' && resource === 'posts') {
         if (params.data.pictures && params.data.pictures.length) {
-            return Promise.all(params.data.pictures.map(convertFileToBase64))
-                .then(base64Pictures => requestHandler(type, resource, {
+            // only freshly dropped pictures are instance of File
```
Comment (jpetitcolas) [inline example/addUploadFeature.js]: Done

**Comment by jpetitcolas** [comment]
Code review applied.

**Comment by fzaninotto** [comment]
tests fail

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**